src/mkdocstrings/extension.py

```diff
@@ -231,17 +231,26 @@ def _process_block(
 
 class _PostProcessor(Treeprocessor):
     def run(self, root: Element) -> None:
+        self.remove_duplicated_headings_from_parent(root)
+
+    def remove_duplicated_headings_from_parent(self, parent: Element) -> bool:
         carry_text = ""
-        for el in reversed(root):  # Reversed mainly for the ability to mutate during iteration.
+        found = False
+        for el in reversed(parent):
```

Comment:
Lets keep the previous comment:

```suggestion
        for el in reversed(parent):  # Reversed mainly for the ability to mutate during iteration.
```